ybn/operators.py

```diff
@@ -123,21 +123,26 @@ def execute(self, context):
             return {"CANCELLED"}
 
         bound_child_type = context.scene.bound_child_type
-        apply_default_flags = context.scene.composite_apply_default_flag_preset
         do_center = context.scene.center_composite_to_selection
 
+        # Always apply the selected flag preset
+        index = context.window_manager.sz_flag_preset_index
+        load_flag_presets()
+        try:
+            preset = flag_presets.presets[index]
+        except IndexError:
+            self.report({"WARNING"}, "Flag preset does not exist or is not selected.")
+            return {"CANCELLED"}
+
         if context.scene.create_seperate_composites or len(selected_meshes) == 1:
-            convert_objs_to_composites(
-                selected_meshes, bound_child_type, apply_default_flags)
+            convert_objs_to_composites(selected_meshes, bound_child_type, preset)
         else:
-            composite_obj = convert_objs_to_single_composite(
-                selected_meshes, bound_child_type, apply_default_flags)
+            composite_obj = convert_objs_to_single_composite(selected_meshes, bound_child_type, preset)
```

Comment:
`convert_objs_to_composites` and `convert_objs_to_single_composite` still expect the `apply_default_flags` parameter, you didn't update them to use the preset. They are in boundhelper.py.

You can change them to take the preset index, that then is passed to the `apply_default_flag_preset` function. And rename it to just `apply_flag_preset`.